docs/source/whitepaper.rst

```diff
@@ -335,20 +334,82 @@ Charles can attempt to delay Bob's withdrawal by challenging Bob's rightful clai
+`L1 non-fill proof` from rollup B to rollup A.
 
-.. todo::
-    Exact specification TBD: https://github.com/beamer-bridge/beamer/issues/346
+When called, the fill manager contract on rollup B recomputes the fill hash from the request hash, and fill ID,
+and checks that no fills exists for the corresponding request hash and fill hash. It then submits a proof
+to the outbox of rollup B indicating that the fill hash is invalid, i.e. that the request hash cannot be mapped to the
+fill hash.
+
+Similarly to the filled L1 resolution case, Bob can then trigger a call on L1 to forward this message to rollup A. This
+message will store in the `resolution registry` that the `fill hash` is invalid and make any claim from Charles with
+the corresponding hash an invalid claim.
+
+To make sure the proof arrives in time on rollup A, Bob will need to call the `fill manager` as soon as he notices a
+false claim for a non-filled request. It takes `finalization time of rollup B` after Bob's call to be able to send the
+proof, while the challenge period is defined to be `finalization time of rollup B + challenge period extension`.
+
+In the case where someone challenges Charles on the false claim at the same time as Bob sends the transaction for the
+proof on rollup B, Bob will not be able to challenge Charles and will not receive any financial reward from having sent
+this transaction. The situation being unlikely to happen and the costs of rollup transactions being low, we believe this
+not to be too big of a problem.
+
+The costly L1 transaction can however be sent only when Bob is properly incentivized, that is he is at stake in the
+challenge against Charles.
+
+Non-fill proofs and self challenges
++++++++++++++++++++++++++++++++++++
+
+We explained in a previous part a strategy for honest fillers to counter wrongful self-challenged claims. To be properly
+incentivized for the submission of the non-fill proof on L1, we need to allow Bob to join a self-challenged claim
+initiated by Charles as an additional challenger.
+
+There is no need for Bob to enter the challenge if the claimer is in a losing position, we only allow new challengers if
+the claimer is ahead in the challenge. This is done by a bookkeeping mechanism of bidders and bids. We store in a mapping
+who bid which amount in total and who was the last bidder.
+
+The rationale behind the book keeping is that anybody who entered into a challenge and put in some stake should be
+rewarded when his position wins, depending on the amount they had to put in. When the challenge ends, if the claimer is
+ahead, he will earn the stakes of every challenger. If a challenger is ahead, each non-last challenger earns a value
+equal to their total stake. The stake of the last challenger being only partially covered by the claimer, he will only
+earn `stake claimer - stake other challengers`, i.e. the remaining tokens.
```

Comment:
I assume your intention here was to make this monospace, not italic, right?
If so, could you please use \`\`, instead of \`, not just here, but everywhere else?

Reply:
Makes sense